pmm-app/tests/pages/qanPage.js

```diff
@@ -129,43 +131,42 @@ module.exports = {
 
   getFilterLocator(filterValue) {
     const filterLocator =
-      "//section[@class='aside__filter-group']//span[contains(text(), '" +
-      filterValue +
-      "')]/../span[@class='checkbox-container__checkmark']";
+     "//section[@class='aside__filter-group']//span[contains(text(), '" +
+     filterValue +
+     "')]/../span[@class='checkbox-container__checkmark']";
     return filterLocator;
   },
 
-  applyFilter(filterValue) {
+  applyFilter(filterValue){
     const filterLocator = this.getFilterLocator(filterValue);
     I.waitForElement(filterLocator, 30);
     I.click(filterLocator);
     I.waitForVisible(this.fields.table, 30);
   },
 
+  async expandAllFilter() {
+    for (let i = 0; i < 4; i++) {
+      // eslint-disable-next-line max-len
+      const numOfElements = await I.grabNumberOfVisibleElements("//section[@class='aside__filter-group']//span[contains(text(), '" + this.filterGroups[i] + "')]/../button[contains(text(), 'See all')]");
+      if (numOfElements === 1) {
+        // eslint-disable-next-line max-len
+        I.click("//section[@class='aside__filter-group']//span[contains(text(), '" + this.filterGroups[i] + "')]/../button[contains(text(), 'See all')]");
```

Comment:
Please move generated selectors to locators functions and use template strings instead of concatenation everywhere

Reply:
const selector = `"//section[@class='aside__filter-group']//span[contains(text(), '${this.filterGroups[i]}')]/../button[contains(text(), 'See all')]");`;
I.click(selector);

Reply:
Done thanks